### PR TITLE
Miner faucet tweaks

### PR DESCRIPTION
**Pull request by darl3n3**

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -352,27 +364,78 @@ func serveFaucet(c *gin.Context) {
...
+		}
+
+		// Calculate what the user is owed
+		//
+		//     [min: 2000, max: (400 * miner_size_TiB / elapsed_days)]
+		//
+		daysSinceLastGrant := uint64(time.Now().Sub(user.MostRecentFaucetGrant).Hours()) / 24
+		if daysSinceLastGrant == 0 {
+			daysSinceLastGrant = 1
+		}
```
Comment (Schwartz10) [inline server.go:423]: Not sure I understand the logic for this. If we follow the approach I outlined in the main thread, we should just be able to just if `requestIsTooSoon` and then `daysSinceLastGrant` will never be 0?

**Comment by Schwartz10** [comment]
As a side note, I'm a little concerned now about https://github.com/openworklabs/filecoin-verifier/issues/9 - i wonder if some of our scan operations aren't working right?

File: lotus.go [hunk 2]
```diff
@@ -284,7 +285,7 @@ func lotusEstimateGasLimit(ctx context.Context, api api.FullNode, msg *types.Mes
 	if err != nil {
 		return 0, err
 	}
-
+	fmt.Println("Gas limit ~>", gasLimit)
```
Comment (Schwartz10) [inline lotus.go:288]: Can remove

File: server.go [hunk 3]
```diff
@@ -20,6 +21,16 @@ import (
...
+	addrs, err := lotusListMiners()
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println("Miners:")
+	for _, addr := range addrs {
+		fmt.Println("  -", addr.String())
+	}
+
```
Comment (Schwartz10) [inline server.go:33]: Can remove

File: server.go [hunk 4]
```diff
@@ -382,12 +445,16 @@ func serveFaucet(c *gin.Context) {
 
 	// Respond to the HTTP request
 	type Response struct {
-		Cid string `json:"cid"`
+		Cid  string `json:"cid"`
+		Sent string `json:"sent"`
```
[the remark this answers is not in the document]
Comment (darl3n3) [reply server.go:449]: Figured it could be useful output on the frontend

File: lotus.go [hunk 5]
```diff
@@ -332,6 +332,53 @@ func lotusSendFIL(ctx context.Context, fromAddr, toAddr address.Address, filAmou
 	return sm.Cid(), nil
 }
 
+func lotusGetMinerInfo(ctx context.Context, addr address.Address) (_ *api.MinerPower, _ address.Address, err error) {
+	defer withStack(&err)
```
[the remark this answers is not in the document]
Comment (darl3n3) [reply lotus.go:336]: Yep

File: aws.go [hunk 6]
```diff
@@ -26,6 +26,15 @@ type AccountData struct {
 	CreatedAt time.Time `json:"created_at"`
 }
 
+func (user User) HasAccountOlderThan(threshold time.Duration) bool {
```
[the remark this answers is not in the document]
Comment (darl3n3) [reply aws.go:29]: Oh, that param should be used.  Will fix

**Pull request closed without merging**